tests/foreman/ui/test_ansible.py

```diff
@@ -448,6 +448,52 @@ def test_positive_assign_and_remove_ansible_role_to_host(self, target_sat, funct
                 == 'No roles assigned directly to the host'
             )
 
+    def test_positive_assign_and_remove_role_to_hostgroup(
+        self, target_sat, module_org, module_location
+    ):
+        """Add and remove functionality for ansible roles in hostgroup
+
+        :id: 5d94a484-92c1-4387-ab92-0649e4c4f907
+
+        :steps:
+            1. Import all roles available by default.
+            2. Create a hostgroup
+            3. Assign hostgroup to the host
+            4. Register a RHEL host to Satellite with given hostgroup.
+            5. Navigate to the new UI for the given hostgroup.
+            6. Add an ansible roles to the hostgroup
+            7. Remove the added ansible roles from the hostgroup
+
+        :expectedresults: The Ansible Role is successfully assign/add/remove from the hostgroup, and no longer shows
+            up on the UI
+        """
+        SELECTED_ROLE = 'RedHatInsights.insights-client'
+        hostgroup = target_sat.api.HostGroup(
+            location=[module_location],
+            organization=[module_org],
+        ).create()
+        host = target_sat.api.Host(
+            organization=module_org, location=module_location, hostgroup=hostgroup
+        ).create()
+        location = host.location.read()
```

Comment:
You don't need read here. You can just use module_location.name directly.

Reply:
all the changes has been done.